lizard.py

```diff
@@ -759,8 +766,14 @@ def get_warnings(code_infos, option):
 
 def print_result(result, option, scheme):
     result = print_and_save_modules(result, option.extensions, scheme)
+    warning_count = 0
+    warning_nloc = 0
     warnings = get_warnings(result, option)
-    warning_count, warning_nloc = print_warnings(option, scheme, warnings)
+    if list(warnings):
+        warnings = get_warnings(result, option)
```

Comment:
@mdmedley Thanks for explaining. That was subtle without looking into definition of ``get_warnings`` then ``whitelist_filter``, both of which, sadly, are undocumented.